code/addons/themes/postinstall.js

```diff
@@ -1,17 +1,29 @@
 const { spawn } = require('child_process');
 
 const PACKAGE_MANAGER_TO_COMMAND = {
-  npm: 'npx',
-  yarn1: 'npx',
-  yarn2: 'yarn dlx',
-  pnpm: 'pnpm dlx',
+  npm: ['npx'],
+  pnpm: ['pnpm', 'dlx'],
+  yarn1: ['npx'],
+  yarn2: ['yarn', 'dlx'],
 };
 
-module.exports = function postinstall(options) {
-  const command = PACKAGE_MANAGER_TO_COMMAND[options.packageManager];
+const selectPackageManagerCommand = (packageManager) => PACKAGE_MANAGER_TO_COMMAND[packageManager];
 
-  spawn(command, ['@storybook/auto-config', 'themes'], {
+const spawnPackageManagerScript = async (packageManager, args) => {
+  const [command, ...baseArgs] = selectPackageManagerCommand(packageManager);
+
+  await spawn(command, [...baseArgs, ...args], {
     stdio: 'inherit',
     cwd: process.cwd(),
+    shell: true,
   });
 };
+
+module.exports = async function postinstall({ packageManager = 'npm' }) {
+  try {
+    await spawnPackageManagerScript(packageManager, ['@storybook/auto-config', 'themes']);
+  } catch (e) {
+    console.error(e);
```

Comment:
what's the difference between logging the error + exiting vs just executing the function and let it fail and bubble the error up?

Reply:
Oops! that was mostly just for my local testing